ffi/utf8proc.lua

```diff
@@ -0,0 +1,63 @@
+--[[--
+Module for utf8 string operations
+
+This is a LuaJIT FFI wrapper for utf8proc.
+]]
+
+local ffi = require("ffi")
+local C = ffi.C
+local BB = require("ffi/blitbuffer")
+
+require("ffi/posix_h")
+require("ffi/utf8proc_h")
+
+local turbojpeg
+if ffi.os == "Windows" then
+    libutf8proc = ffi.load("libs/libutf8proc.dll")
+elseif ffi.os == "OSX" then
+    libutf8proc = ffi.load("libs/libutf8proc.dylib")
+else
+    libutf8proc = ffi.load("libs/libutf8proc.so.2")
+end
+
+local Utf8Proc = {}
+
+function Utf8Proc.lowercase(str)
+    local str8_len = str:len()
+
+    local str8 = ffi.new("uint8_t[" .. str8_len .."]")
+    local str32 = ffi.new("int32_t[" .. str8_len .. "]")
+
+    -- copy string contents to uint8_t array
+    for i = 1,str8_len do
+        str8[i-1] = str:byte(i)
+    end
```

Comment:
:?

(Lua strings are immutable, you can use em as a `const char*`).

If you *do* need it mutable, there's `ffi.string` if you nee dit in Lua, or just `ffi.memcpy` in C?